google_gke/cluster.tf

```diff
@@ -0,0 +1,206 @@
+# TBD block of things to investigate further in the future:
+# * confidential nodes (basically encrypted in-use memory on VMs underlying nodes; in beta)
+# * GKE sandbox (basically host kernel protection on nodes; in beta)
+# * cluster telemetry (some kinda new monitoring / logging / metrics aggregation & dashboard for gke clusters; in beta)
+# * enable_binary_authorization (all container images validated by Google Binary Authorization; needs further impact investigation)
+# * enable_l4_ilb_subsetting (needs further impact investigation)
+# * shielded_instance_config.enable_secure_boot & shielded_instance_config.enable_integrity_monitoring (needs further impact investigation)
+# * database_encryption to be added with CloudKMS key (postponed for adding CloudKMS keys structure to Terraform or secrets management)
+# * authenticator_groups_config (postponed)
+
+#
+# GKE Cluster
+#
+resource "google_container_cluster" "primary" {
+  provider = google-beta
+
+  name            = local.cluster_name
+  description     = var.description
+  location        = var.region
+  project         = local.project_id
+  resource_labels = local.labels
+
+  release_channel {
+    channel = var.release_channel
+  }
+
+  workload_identity_config {
+    workload_pool = "${local.project_id}.svc.id.goog"
+  }
+
+  # Networking: Defaulting to Shared VPC Setup
+  network         = local.network
+  subnetwork      = local.subnetwork
+  networking_mode = "VPC_NATIVE"
+  ip_allocation_policy {
+    cluster_secondary_range_name  = local.pods_ip_cidr_range_name
+    services_secondary_range_name = local.services_ip_cidr_range_name
+  }
+  master_auth {
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+  master_authorized_networks_config {
+    dynamic "cidr_blocks" {
+      for_each = var.master_authorized_networks
+
+      content {
+        cidr_block   = lookup(cidr_blocks.value, "cidr_block", "")
+        display_name = lookup(cidr_blocks.value, "display_name", "")
+      }
+    }
+  }
+
+  # are we really using gke dataplane v2 (with limitations) over calico?
```

Comment:
This piece is pretty confusing for me on how the terraform provider works, but I *think* we will just be using calico unless we actually set:
```
datapath_provider = "ADVANCED_DATAPATH"
```

Other than the scalability promised by dataplane v2, I'm not sure if we actually need it. However it looks like it's something that needs to be enabled at cluster creation so if we want it we should probably make the decision before standing up new clusters. The terraform provider docs are extremely lacking on this topic from what I've seen.

Reply:
Are you confusing dataplane/calico and cilium / the data paths? Otherwise, not following what you're trying to understand here.

Reply:
I changed the networking policy setup from Calico to this upon your motivation to use dataplane.

Reply:
I definitely might be. It looks like to enable dataplane v2 you need to **not** specify the provider (like you did here) AND set the `datapath_provider`. I  haven't tested this yet though, so maybe that's a default now? Just looking at [this comment](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/656#issuecomment-910108440) and [this example](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google/latest/examples/simple_regional_beta). 

When we had talked before the docs seem to indicate that dataplane v2 was "the new way" GKE was going to push people, but I wonder if that's not necessarily the case. Are there use-cases for calico that you think would make it worth using over dataplane v2? I'm honestly not that opinionated either way, it's just something that looks like it has to be set at cluster creation.

Reply:
chatted with @moz-astults , have a list of questions to answer here. going to make the dataplane multi-field enablement a TF module variable so we can test.

Reply:
this is a variable now, will work as a follow-up (which this was always listed as for this PR) to confirm what astults wants and what is needed to migrating existing network policy usage.